File: TaskManager/labFinal_164/model/Issue.java
package labFinal_164.model;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

public class Issue {
    private String id;
    private String title;
    private String description;
    private Priority priority;
    private Status status;
    private LocalDateTime creationDate;
    private LocalDateTime lastModifiedDate;
    private String assignedTo;
    private List<Comment> comments = new ArrayList<>();

    public Issue(String id, String title, String description, Priority priority ) {
        this.id = id;
        this.title = title;
        this.description = description;
        this.priority = priority;
        this.status = Status.OPEN;
        this.creationDate = LocalDateTime.now();
        this.lastModifiedDate = creationDate;

    }


    public String getId() {
        return id;
    }

    public void setStatus(Status status) { this.status = status; }
    public void addComment(Comment comment) { comments.add(comment); }

    public Issue deepClone() {
        Issue cloned = new Issue(this.id, this.title, this.description, this.priority);
        cloned.status = this.status;
        cloned.comments = new ArrayList<>(this.comments); // Shallow copy of immutable comments
        return cloned;
    }

    public String getTitle() {
        return title;
    }

    public Status getStatus() {
        return status;
    }

    public List<Comment> getComment() {
        return comments;
    }

}

File: TaskManager/TaskManager_210042112/TaskManager_210042112/Issue.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Xml.Linq;

namespace TaskManager_210042112
{
    public class Issue
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        
        public DateTime CreationDate { get; set; }
        public DateTime LastModifiedDate { get; set; }
        public string AssignedTo { get; set; }
        

        public Issue(int id, string title, string description)
        {
            Id = id;
            Title = title;
            Description = description;
           
            CreationDate = DateTime.Now;
            LastModifiedDate = DateTime.Now;
       
            
        }
    }
}
